clang/docs/ReleaseNotes.rst

```diff
@@ -120,6 +120,10 @@ Non-comprehensive list of changes in this release
 New Compiler Flags
 ------------------
 
+- ``-Wmissing-designated-field-initializers``, grouped under ``-Wmissing-designated-field-initializers``.
```

Comment:
Would it be ok to also add this patch to the 18.x release branch?
It would be nice to have a way to keep old behavior of the m-f-i check when upgrading from Clang 16/17 to 18.

Reply:
CC @AaronBallman

Reply:
I think it would be good to have if we are certain the changes are correct and safe (we're running out of time for getting things into 18.x).

Reply:
```suggestion
- ``-Wmissing-designated-field-initializers``, grouped under ``-Wmissing-field-initializers``.
```